File: Hospital Information System_ui/src/app/Filters/PatientFilter.ts
import { PaginationFilter } from '../Filters/PaginationFilter'


export class PatientFilter extends PaginationFilter {


    name: string;
    fileNo: number;
    phoneNumber: string;


    
    constructor(pageNumber: Number,
        pageSize: Number,
        name: string,
        fileNo: number,
        phoneNumber: string) {

        super(pageNumber, pageSize);

        this.name = name;
        this.fileNo = fileNo;
        this.phoneNumber = phoneNumber;
    }

    reset(){
        this.pageNumber = 1;
        this.pageSize = 10;
        this.name = "";
        this.fileNo = 0;
        this.phoneNumber = "";
    }

}

File: Hospital Information System API/HospitalInformationSystemAPI/Filters/PatientFilter.cs
using HospitalInformationSystemAPI.Models;
using Microsoft.EntityFrameworkCore;
using System.Linq;

namespace HospitalInformationSystemAPI.Filters
{
    public class PatientFilter : PaginationFilter, IFilter<Patient>
    {
        public string Name { get; set; }
        public int FileNo { get; set; }
        public string PhoneNumber { get; set; }

        public PatientFilter() { }
        public PatientFilter(PatientFilter filter) : base(filter.PageNumber, filter.PageSize)
        {
            this.Name = filter.Name;
            this.FileNo = filter.FileNo;
            this.PhoneNumber = filter.PhoneNumber;
        }


        public IQueryable<Patient> Filter(IQueryable<Patient> data)
        {
            validate();

            if (!string.IsNullOrWhiteSpace(Name))
                data = data.Where(p => p.Name.ToUpper().Contains(Name.ToUpper()));

            if (FileNo > 0)
                data = data.Where(p => p.FileNo == FileNo);

            if (!string.IsNullOrWhiteSpace(PhoneNumber))
                data = data.Where(p => p.PhoneNumber.ToUpper().Contains(PhoneNumber.ToUpper()));

            return Paginate(data);
        }

        public IQueryable<Patient> Filter(DbSet<Patient> data)
        {
            return Filter(data.AsQueryable<Patient>());
        }
    }
}
